### PR TITLE
Adds PopupMessage to being hugged [Fix  #7644]

### LINKED ISSUE #7644: Hugging is only visible clientside
## Description

Simple as the title suggests. The hugging action can only be seen from the client side perspective. It's not a big deal, but it just is kind of weird. This is probably due to repurposed petting code.
**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by Mirino97**
## About the PR 
Fix  #7644
It's my understanding that InteractionPopupSystem does not allow for the propagation of a DIFFERENT message to surrounding clients (besides the one I've previously set to "**You** hug X", so I've decided to add it to the System. Now, I COULD change the initial message from "You hug X" to "X hugged Y" and use the already implemented broadcasting system, but I thought that keeping both messages would be better. I could also add "X hugs you" to the receiving client if asked to, this PR is more to feel the room about this change.

Feel free to go into detail about anything you want to (like suggestions, asking to change the whole methods, yada yada), I'm very new to C# and to a project this large, so it's absolutely fair to say I'm lost and just doing... whatever I think is best.

**Screenshots**

Screenshot 2022-04-21 222324

**Changelog**

:cl:
- tweak: You will now see when someone hugs you!

### CODE REVIEW

**Comment by metalgearsloth** [comment]
mission critical PR

**Comment by Mirino97** [comment]
> mission critical PR

Honestly how have we gone so long without this?

**Comment by DogZeroX** [comment]
Being unable to see who is hugging me gave me depression.

File: Content.Server/Interaction/Components/InteractionPopupComponent.cs [hunk 1]
```diff
@@ -56,6 +56,12 @@ public sealed class InteractionPopupComponent : Component
         [DataField("popupPerceivedByOthers")]
         public bool PopupPerceivedByOthers = false;
 
+        /// <summary>
+        /// If the popup is perceived by others, which will it be?
+        /// </summary>
+        [DataField("messagePerceivedByOthers")]
+        public string? MessagePerceivedByOthers;
```
Comment (metalgearsloth) [inline Content.Server/Interaction/Components/InteractionPopupComponent.cs:63]: IMO PopupPerceivedByOthers should be nuked if this is added as they double-up.

File: Content.Server/Interaction/InteractionPopupSystem.cs [hunk 2]
```diff
@@ -57,7 +57,16 @@ private void OnInteractHand(EntityUid uid, InteractionPopupComponent component,
...
-            _popupSystem.PopupEntity(msg, uid, Filter.Pvs(uid)); //play for everyone in range
+        {
+            if (component.MessagePerceivedByOthers != null)
+            {
+                string msgOthers = Loc.GetString(component.MessagePerceivedByOthers,("user", args.User), ("target", uid));
+                _popupSystem.PopupEntity(msg, uid, Filter.Entities(args.User));
+                args.User.PopupMessageOtherClients(msgOthers);
+            }
+            else
+                _popupSystem.PopupEntity(msg, uid, Filter.Pvs(uid)); //play for everyone in range
```
Comment (metalgearsloth) [inline Content.Server/Interaction/InteractionPopupSystem.cs:68]: pass in entitymanager to filters plz.

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**